examples/linear_model/plot_logistic_multinomial_rcv1.py

```diff
@@ -0,0 +1,101 @@
+"""
+=====================================================
+Multiclass logisitic regression on newgroups20
+=====================================================
+
+Comparison of multinomial logistic L1 vs one-versus-rest L1 logistic regression
+to classify documents from the newgroups20 dataset. Multinomial logistic
+regression yields more accurate results and is faster to train on the larger
+scale dataset.
+"""
+import time
+
+import matplotlib.pyplot as plt
+import numpy as np
+
+from sklearn.datasets import fetch_20newsgroups_vectorized
+from sklearn.linear_model import LogisticRegression
+from sklearn.model_selection import train_test_split
+
+print(__doc__)
+# Author: Arthur Mensch
+
+t0 = time.clock()
+
+# Turn down for faster run time
+n_samples = 10000
+
+# Memorized fetch_rcv1 for faster access
+dataset = fetch_20newsgroups_vectorized('all')
+X = dataset.data
+y = dataset.target
+X = X[:n_samples]
+y = y[:n_samples]
+
+X_train, X_test, y_train, y_test = train_test_split(X, y,
+                                                    random_state=42,
+                                                    stratify=y,
+                                                    test_size=0.1)
+train_samples, n_features = X_train.shape
+n_classes = np.unique(y).shape[0]
+
+print('Dataset 20newsgroup, train_samples=%i, n_features=%i, n_classes=%i'
+      % (train_samples, n_features, n_classes))
+
+models = {'ovr': {'name': 'One versus Rest'},
+          'multinomial': {'name': 'Multinomial'}}
+
+for model in models:
+    solver = 'saga' if model == 'multinomial' else 'liblinear'
+    # Add initial chance-level values for plotting purpose
+    accuracies = [1 / n_classes]
+    times = [0]
+    sparsities = [1]
+
+    # Small number of epochs for fast runtime
+    for this_max_iter in [1, 3]:
+        print('[model=%s, solver=%s] Number of epochs: %s' %
+              (model, solver, this_max_iter))
+        lr = LogisticRegression(solver=solver,
+                                multi_class=model,
+                                C=1,
+                                penalty='l1',
+                                fit_intercept=False,
+                                max_iter=this_max_iter,
+                                random_state=42,
+                                )
+        t1 = time.clock()
+        lr.fit(X_train, y_train)
+        train_time = time.clock() - t1
+
+        y_pred = lr.predict(X_test)
+        accuracy = np.sum(y_pred == y_test) / y_test.shape[0]
+        sparsity = np.mean(lr.coef_ == 0) * 100
+        accuracies.append(accuracy)
+        sparsities.append(sparsity)
+        times.append(train_time)
+    models[model]['times'] = times
+    models[model]['sparsities'] = sparsities
+    models[model]['accuracies'] = accuracies
+    print('Test accuracy for model %s: %.4f' % (model, accuracies[-1]))
+    print('Model sparsity for model %s: % .2f' % (model, sparsities[-1]))
```

Comment:
Maybe you could print the number of of non-zero weights selected for each class label.